test/server/routes/user.test.js

```diff
@@ -210,3 +212,143 @@ describe('Update user profile', () => {
       });
   });
 });
+
+let bookTitle;
+describe('Fetch Books router', () => {
+  it('It should get book with query string page and limit', (done) => {
+    chai.request(server)
+      .get(`${BOOKS_BASE_URL}?page=1&limit=10`)
+      .end((error, response) => {
+        expect(response).to.have.status(200);
+        expect(response.body).to.be.an('object');
+        expect(response.body.message).to.equal('request successful');
+        done();
+      });
+  });
+  it('It should get book without query string page and limit', (done) => {
+    chai.request(server)
+      .get(`${BOOKS_BASE_URL}`)
+      .end((error, response) => {
+        bookTitle = response.body.books[0].title.slice(0, 7);
+        expect(response).to.have.status(200);
+        expect(response.body).to.be.an('object');
+        expect(response.body.message).to.equal('request successful');
+        done();
+      });
+  });
+  it('It should get response:"page does not exist" if current page is greater than pages', (done) => {
+    chai.request(server)
+      .get(`${BOOKS_BASE_URL}?page=10000000&limit=10`)
+      .end((error, response) => {
+        expect(response).to.have.status(400);
+        expect(response.body).to.be.an('object');
+        expect(response.body.message).to.equal('page does not exist');
+        done();
+      });
+  });
+  it('It should get book wrong query string page and limit', (done) => {
+    chai.request(server)
+      .get(`${BOOKS_BASE_URL}?page=ed&limit=lcd`)
+      .end((error, response) => {
+        expect(response).to.have.status(400);
+        expect(response.body.errors).to.be.an('object');
+        expect(response.body.errors.query.page).to.be.equal('page value must be at least 1 and an integer');
+        expect(response.body.errors.query.limit).to.be.equal('limit value must be at least 1 and an integer');
+        done();
+      });
+  });
+});
+
+describe('Search Books', () => {
+  it('should search a book based on the book title', (done) => {
+    chai.request(server)
+      .get('/api/v1/books')
+      .query({ title: bookTitle })
+      .end((error, response) => {
+        expect(response).to.have.status(200);
+        expect(response.body.status).to.equal('success');
+        expect(response.body.message).to.equal('request successful');
+        expect(response.body.books).to.be.an('array');
+        expect(response.body.books.length).to.not.equal(0);
+        done();
+      });
+  });
+
+  it('should search a book based on the tag', (done) => {
+    chai.request(server)
+      .get('/api/v1/books')
```

Comment:
Duplicate declaration of routes, you can initialize above and call declarations below

Reply:
seconded

Reply:
Done